### PR TITLE
Inject sitemap builder into sitemap loader, fixes #31

### LINKED ISSUE #31: Symfony 3.2 deprecation
> User Deprecated: The "sylius.sitemap_builder"  service is private, getting it from the container is deprecated since  Symfony 3.2 and will fail in 4.0. You should either make the service  public, or stop using the container directly and use dependency  injection instead.

I believe caused by `$providers = $this->container->get('sylius.sitemap_builder')->getProviders();` at row 41 in SitemapLoader.php.

I suggest us to inject the `sylius.sitemap_builder` service into the `sylius.sitemap.routing_loader` service, no need to have it public that way.

### PR DESCRIPTION
**Pull request by sweoggy**

### CODE REVIEW

**Comment by sweoggy** [comment]
Ready for review @stefandoorn

File: src/Routing/SitemapLoader.php [hunk 1]
```diff
@@ -25,10 +25,21 @@ class SitemapLoader extends Loader implements ContainerAwareInterface
      */
     private $loaded = false;
 
+    /**
+     * @var SitemapBuilderInterface
+     */
+    private $sitemapBuilder;
+
+    public function __construct(SitemapBuilderInterface $sitemapBuilder)
```
Comment (stefandoorn) [inline src/Routing/SitemapLoader.php]: Could you add a docblock in here?

File: src/Routing/SitemapLoader.php [hunk 2]
```diff
@@ -38,7 +49,7 @@ public function load($resource, $type = null): RouteCollection
 
         $routes = new RouteCollection();
 
-        $providers = $this->container->get('sylius.sitemap_builder')->getProviders();
+        $providers = $this->sitemapBuilder->getProviders();
```
Comment (stefandoorn) [inline src/Routing/SitemapLoader.php]: While we are improving things anyway, I would opt to move this to combine it at once in line 54. No really need to put it in a separate variable anyway.

**Comment by stefandoorn** [comment]
Looks good to me. Can you rebase to a single commit? Then I think it's good to merge :-)

**Comment by sweoggy** [comment]
@stefandoorn done

**Comment by stefandoorn** [comment]
Thanks!

**Pull request merged**